apps/src/gamelab/GameLab.js

```diff
@@ -1423,36 +1479,10 @@ GameLab.prototype.onP5Draw = function () {
   if (this.JSInterpreter && this.eventHandlers.draw) {
     this.drawInProgress = true;
     if (getStore().getState().runState.isRunning) {
-      this.eventHandlers.draw.apply(null);
-      if (this.level.validationCode) {
-        try {
-          const validationResult =
-            this.JSInterpreter.interpreter.marshalInterpreterToNative(
-              this.JSInterpreter.evalInCurrentScope(`
-                (function () {
-                  validationState = null;
-                  validationResult = null;
-                  ${this.level.validationCode}
-                  return {
-                    state: validationState,
-                    result: validationResult
-                  };
-                })();
-              `)
-            );
-          if (validationResult.state === 'succeeded') {
-            console.log('success!');
-            const testResult = validationResult.result ||
-                TestResults.ALL_PASS;
-            this.onPuzzleComplete(false, testResult);
-          } else if (validationResult === 'failed') {
-            // TODO(ram): Show failure feedback
-          }
-        } catch (e) {
-          // If validation code errors, assume it was neither a success nor failure
-          console.error(e);
-        }
-      }
+      this.measureDrawLoop(DRAW_LOOP_MEASURE, () => {
```

Comment:
Are you confident this won't affect Game Lab perf?

Reply:
Did some basic testing, and found that as long as I clear the previous marks every loop, its performance impact is negligible. A million mark/measure calls take ~3 seconds.
![image](https://user-images.githubusercontent.com/1070243/43668840-ed3a715c-9733-11e8-9ee5-dd96e234437a.png)